### PR TITLE
Tweaks soda can suicide

### PR DESCRIPTION
**Pull request by 81Denton**
Probably my finest PR so far

ssssip

### CODE REVIEW

**Comment by PKPenguin321** [comment]
second one should be `They don't make 'em like Orion Trail anymore.

**Comment by 81Denton** [comment]
Good idea, I have a few others on my mind too

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 1]
```diff
@@ -402,11 +402,12 @@
...
 /obj/item/reagent_containers/food/drinks/soda_cans/suicide_act(mob/living/carbon/user)
-	user.visible_message("<span class='suicide'>[user] is trying to eat \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
-	var/obj/item/trash/can/crushed_can = new /obj/item/trash/can(user.loc)
-	crushed_can.icon_state = icon_state
-	qdel(src)
-	return BRUTELOSS
+	user.visible_message("<span class='suicide'>[user] takes a big sip! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(user.loc,'sound/items/drink.ogg', 80, 1)
+	reagents.clear_reagents() //a big sip
+	user.say(pick("Now, Outbomb Cuban Pete, THAT was a game.", "They don't make 'em like Orion Trail anymore.", "You know what they say. Worst day of spess carp fishing is better than the best day at work.", "They don't make 'em like Singularity engines anymore.'"))
```
Comment (PKPenguin321) [inline code/modules/food_and_drinks/drinks/drinks.dm]: `They don't make 'em like Singularity engines anymore.'`
bad apostrophe and capitalization

Comment (81Denton) [reply code/modules/food_and_drinks/drinks/drinks.dm]: English capitalization rules are the devil

**Comment by 81Denton** [comment]
Alright, now the suicide_act checks if A) the can is open and B) if it contains soda. If it's closed, you'll open it before sipping. If it's empty, you'll fall to the ground in shame.

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 2]
```diff
@@ -402,11 +402,18 @@
...
+		return SHAME
+	if(!is_drainable())
+		open_soda()
+		sleep(10)
+	user.visible_message("<span class='suicide'>[user] takes a big sip from [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(user.loc,'sound/items/drink.ogg', 80, 1)
+	reagents.clear_reagents() //a big sip
+	user.say(pick("Now, Outbomb Cuban Pete, THAT was a game.", "They don't make em like Orion Trail anymore.", "You know what they say. Worst day of spess carp fishing is better than the best day at work.", "They don't make em like singularity engines anymore.'"))
+	sleep(20) //dramatic pause
+	return TOXLOSS
```
Comment (Iamgoofball) [inline code/modules/food_and_drinks/drinks/drinks.dm]: this should check your age, if you're over 40 you survive

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 3]
```diff
@@ -402,11 +402,18 @@
...
-	return BRUTELOSS
+	if(!reagents.total_volume)
+		user.visible_message("<span class='warning'>[user] is trying to take a big sip from [src]... The can is empty!</span>")
+		return SHAME
+	if(!is_drainable())
+		open_soda()
+		sleep(10)
+	user.visible_message("<span class='suicide'>[user] takes a big sip from [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(user.loc,'sound/items/drink.ogg', 80, 1)
+	reagents.clear_reagents() //a big sip
```
Comment (Iamgoofball) [inline code/modules/food_and_drinks/drinks/drinks.dm]: transfer all the reagents into the mob instead

File: code/modules/food_and_drinks/drinks/drinks.dm (hunk 2 above)
Comment (81Denton) [reply code/modules/food_and_drinks/drinks/drinks.dm]: this is the level of immersion I strive for

Comment (81Denton) [reply code/modules/food_and_drinks/drinks/drinks.dm]: what would be the best way to handle it? plain `return` just calls the unarmed suicide_act()

**Comment by 81Denton** [comment]
Players 40 years and older now survive the big sip. Reagents get transfered to the mob too, so the soda doesn't magically vanish.

**Comment by obscolene** [comment]
>Players 40 years and older now survive the big sip. Reagents get transfered to the mob too, so the soda doesn't magically vanish.

Add: You can now use soda cans as improvised 30u pills.

**Comment by PKPenguin321** [comment]
Yeah that seems a tad exploitable

**Comment by Iamgoofball** [comment]
@PKPenguin321 you could also just spam press page down to achieve the same effect, it's really not an issue

**Comment by Iamgoofball** [comment]
@81Denton don't worry about it quite frankly think of the downside, everyone on the station now has to be a 40 year old boomer

**Comment by 81Denton** [comment]
The best compromise would probably be `H.Paralyze(50)` while the boomer is sipping and contemplating

**Comment by Iamgoofball** [comment]
https://github.com/81Denton/tgstation/pull/7

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 4]
```diff
@@ -401,12 +401,26 @@
...
+		sleep(10)
+	H.visible_message("<span class='suicide'>[H] takes a big sip from [src]! It looks like [H.p_theyre()] trying to commit suicide!</span>")
+	playsound(H.loc,'sound/items/drink.ogg', 80, 1)
+	reagents.trans_to(H, src.reagents.total_volume) //a big sip
+	sleep(5)
+	H.say(pick("Now, Outbomb Cuban Pete, THAT was a game.", "All these new fangled arcade games are too slow. I prefer the classics.", "They don't make 'em like Orion Trail anymore.", "You know what they say. Worst day of spess carp fishing is better than the best day at work.", "They don't make 'em like good old fashioned singularity engines anymore."))
+	if(H.age >= 40)
+		H.Stun(50)
+		sleep(50)
+		playsound(H.loc,'sound/items/drink.ogg', 80, 1)
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: remove .loc

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 5]
```diff
@@ -401,12 +401,26 @@
...
-	return BRUTELOSS
+/obj/item/reagent_containers/food/drinks/soda_cans/suicide_act(mob/living/carbon/human/H)
+	if(!reagents.total_volume)
+		H.visible_message("<span class='warning'>[H] is trying to take a big sip from [src]... The can is empty!</span>")
+		return SHAME
+	if(!is_drainable())
+		open_soda()
+		sleep(10)
+	H.visible_message("<span class='suicide'>[H] takes a big sip from [src]! It looks like [H.p_theyre()] trying to commit suicide!</span>")
+	playsound(H.loc,'sound/items/drink.ogg', 80, 1)
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: ditto

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by MrDoomBringer** [comment]
@Iamgoofball if you use the stun() proc, im pretty sure things like meth will effect it, causing the proc to fall out of sync. You might wanna address that.

**Pull request merged**